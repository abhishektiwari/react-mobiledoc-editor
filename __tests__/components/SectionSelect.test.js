import React from 'react';
import SectionSelect from '../../src/components/SectionSelect';
import { shallow } from 'enzyme';

describe(SectionSelect, () => {
  it('should render children', () => {
    const wrapper = shallow(<SectionSelect tags={['p']} className="headers" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should add section when new tag is selected', () => {
    const editor = { toggleSection: jest.fn() };
    const context = { editor };
    const wrapper = shallow(<SectionSelect tags={['p']} />, { context });
    wrapper.find('select').simulate('change', { target: { value: 'p' }});
    expect(editor.toggleSection).toBeCalledWith('p');
  });

  it('should remove active section when nothing is selected', () => {
    const editor = { toggleSection: jest.fn() };
    const context = { editor, activeSectionTags: ['p']};
    const wrapper = shallow(<SectionSelect tags={['p']} />, { context });
    wrapper.find('select').simulate('change', { target: { value: "" }});
    expect(editor.toggleSection).toBeCalledWith('p');
  });
});
