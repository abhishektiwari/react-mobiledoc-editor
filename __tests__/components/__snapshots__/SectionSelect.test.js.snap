// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionSelect should render children 1`] = `
<select
  className="headers"
  onChange={[Function]}
  value=""
>
  <option
    value=""
  />
  <option
    key="p"
    value="p"
  >
    P
  </option>
</select>
`;
